src/contracts/oracles/DelayedOracle.sol

```diff
@@ -56,19 +56,33 @@ contract DelayedOracle is IBaseOracle, IDelayedOracle {
 
   /// @inheritdoc IDelayedOracle
   function updateResult() external returns (bool _success) {
-    // Check if the delay passed
-    if (!_delayHasElapsed()) revert DelayedOracle_DelayHasNotElapsed();
     // Read the price from the median
     (uint256 _priceFeedValue, bool _hasValidValue) = _getPriceSourceResult();
-    // If the value is valid, update storage
-    if (_hasValidValue) {
-      // Update state
-      _currentFeed = _nextFeed;
-      _nextFeed = Feed(_priceFeedValue, true);
-      lastUpdateTime = block.timestamp;
-      // Emit event
-      emit UpdateResult(_currentFeed.value, lastUpdateTime);
+
+    // Check if the delay to set the new feed passed
+    if (!_delayHasElapsed()) {
+      // If it hasn't passed, check if the upcoming feed is valid
+      // in the case that it is not valid we check if we can fetch a new valid feed to replace it.
+      if (!_nextFeed.isValid) {
+        // Check if the newly fetched feed is valid
+        if (_hasValidValue) {
+          _nextFeed = Feed(_priceFeedValue, _hasValidValue);
+        }
+        // Store the new next Feed
+        return _hasValidValue;
```

Comment:
hmmmmmmmmmmm
i think we need to update the `lastUpdateTime`
the feed must be available 1hr (or delay time) after the valid feed was fetched (not after the 1st invalid one)